framework/db/mysql/QueryBuilder.php

```diff
@@ -147,6 +147,18 @@ public function checkIntegrity($check = true, $schema = '', $table = '')
     /**
      * @inheritdoc
      */
+    public function buildFrom($tables, &$params)
+    {
+        if (count($tables) < 2) {
```

Comment:
shouldn't it also check if there's `join`? Because if there's no join, then we don't need the parenthesis.